<a href="https://colab.research.google.com/github/rouhuawu/my-first-project/blob/main/notebooks/1-data-exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
print("Hello from my notebook!")

Hello from my notebook!


In [1]:
!git clone https://github.com/rouhuawu/my-first-project.git

Cloning into 'my-first-project'...
remote: Enumerating objects: 23, done.
remote: Counting objects: 100% (23/23), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 23 (delta 2), reused 20 (delta 2), pack-reused 0 (from 0)
Receiving objects: 100% (23/23), 6.14 KiB | 3.07 MiB/s, done.
Resolving deltas: 100% (2/2), done.


In [2]:
!pip install numpy scipy

In [3]:
import sys

# 將 'src' 資料夾的路徑添加到 Python 的搜尋路徑中
sys.path.append('/content/my-first-project/src')

# 載入函式
from solver import create_random_instance, simulated_annealing

print("✅ 成功從 my-first-project/src/solver.py 載入函式！")

✅ 成功從 my-first-project/src/solver.py 載入函式！


In [17]:
print("🚀 開始訓練...")
jobs = create_random_instance(num_jobs=68, num_machines=20)
best_perm, best_makespan = simulated_annealing(jobs, num_machines=10, steps=10000)

print("--- 訓練完成 ---")
print(f"最佳結果 (Best makespan): {best_makespan}")

🚀 開始訓練...
--- 訓練完成 ---
最佳結果 (Best makespan): 871


#### 實驗筆記
jobs = create_random_instance(num_jobs=68, num_machines=20)

best_perm, best_makespan = simulated_annealing(jobs, num_machines=10, steps=10000)

(Best makespan): 871
